smalltalksrc/VMMakerTests/VMSpurOldSpaceGarbageCollectorTest.class.st

```diff
@@ -416,6 +416,24 @@ VMSpurOldSpaceGarbageCollectorTest >> testEphemeronOverflowUnscannedEphemeronQue
 		equals: numberJustOverLimit
 ]
 
+{ #category : #tests }
+VMSpurOldSpaceGarbageCollectorTest >> testEphemeronWithImmediateKeyShouldNotFail [
+
+	| ephemeron1 key |
+	ephemeron1 := self newEphemeronObjectWithSlots: 5.
+	self keepObjectInVMVariable1: ephemeron1.
+	key := memory integerObjectOf: 15.
+	memory storePointer: 0 ofObject: ephemeron1 withValue: key.
+
+	memory setCheckForLeaks: 63. "all"
+	memory fullGC.
+
+	self
+		assert:
+		(memory fetchInteger: 0 ofObject: self keptObjectInVMVariable1)
+		equals: 15
+]
```

Comment:
Why here the assertion says just `15` instead of `key`? @guillep

Reply:
I did a `fetchInteger:`, to make sure the same value is there.
What would be the most readable?
I think in any case we will need a comment, right?

Reply:
Ahhhh okkkkk. I don't think that a comment is needed, I just need to learn how to read 😅 

Maybe having `storeInteger:` also would help with the _symmetry_.